### PR TITLE
post method, please review 

### PR DESCRIPTION
**Pull request by youthtrouble**

### CODE REVIEW

File: handleClipsPost.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+
+func (s *server) handleClipsPost(w http.ResponseWriter, r *http.Request) {
+	// Parse and decode the request body into a new `Clip` instance
+	var clip Clip 
+
+	err := json.NewDecoder(r.Body).Decode(&clip)
+	if err != nil {
+		// If there is something wrong with the request body, return a 400 status
+		s.Error(w, r, err, 400)
```
Comment (tfalamu) [inline handleClipsPost.go]: As a suggestion, we can avoid hard-coding response codes by using the defaults from the http library.
```go
s.Error(w, r, err, http.StatusBadRequest)
```

File: handleClipsPost.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	}
+
+	uid := clip.ID
+	clip.Timestamp = time.Now().UnixNano() 
+
+	if err = s.db.Put(uid, clip); err != nil {
+		// If there is any issue with inserting into the database, return a 500 error
+		s.Error(w, r, err, 500)
```
Comment (tfalamu) [inline handleClipsPost.go]: Here too, we could use http.StatusInternalServerError

File: handleClipsPost.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	// Parse and decode the request body into a new `Clip` instance
+	var clip Clip 
+
+	err := json.NewDecoder(r.Body).Decode(&clip)
+	if err != nil {
+		// If there is something wrong with the request body, return a 400 status
+		s.Error(w, r, err, 400)
+	}
+
+	uid := clip.ID
```
Comment (tfalamu) [inline handleClipsPost.go]: The UID is different from the clip's ID. Think of the UID as the username for the currently logged in user. To obtain the UID in your handler, there's a convenience method called "uidFromContext" defined on the server receiver. Use that to obtain your UID. 
For now, please ignore the clip ID.

File: handleClipsPost.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	clip.Timestamp = time.Now().UnixNano() 
+
+	if err = s.db.Put(uid, clip); err != nil {
+		// If there is any issue with inserting into the database, return a 500 error
+		s.Error(w, r, err, 500)
+	}
+	w.WriteHeader(http.StatusCreated)
+	json.NewEncoder(w).Encode(clip)
```
Comment (tfalamu) [inline handleClipsPost.go]: Awesome, this is what I had in mind too, it's good practice to return the newly created object! :smile:
There is a convenience method defined on the server to respond with json too.
```go
s.JSON(w, r, clip, http.StatusCreated)
```

**Review by tfalamu — CHANGES_REQUESTED** [review]

File: handleClipsPost.go (hunk 2 above)
Comment (youthtrouble) [reply handleClipsPost.go]: change made

File: handleClipsPost.go (hunk 3 above)
Comment (youthtrouble) [reply handleClipsPost.go]: I missed that!, thank you I've made the necessary change

File: handleClipsPost.go (hunk 4 above)
Comment (youthtrouble) [reply handleClipsPost.go]: corrected, Thanks.

**Comment by tfalamu** [comment]
Looks great :rocket:

**Review by tfalamu — APPROVED** [review]

**Pull request merged**